python/examples/atomistic/3-atomistic-model-with-nl.py

```diff
@@ -0,0 +1,300 @@
+r"""
+.. _atomistic-tutorial-nl-md:
+
+Running Molecular Dynamics with a Neighbor List (NL)
+====================================================
+
+.. py:currentmodule:: metatensor.torch.atomistic
+
+This tutorial demonstrates how to use metatensor atomistic models to run a Molecular
+Dynamics (MD) simulation using ASE in combination with a neighbor list.
+
+This tutorial assumes knowledge of how to export an atomistic model and run it with the
+ASE calculator. If you haven't read the corresponding examples, please refer to
+:ref:`atomistic-tutorial-export` and :ref:`atomistic-tutorial-md`.
+
+In this example, we will run a short molecular dynamics simulation of 125 already
+equilibrated liquid argon atoms at a temperature of 94.4 K and a mass density of
+1.374 g/cm³ to obtain the pair-correlation function :math:`g(r)` of
+the liquid.
+"""
+
+# sphinx_gallery_thumbnail_number = 2
+
+from typing import Dict, List, Optional
+
+# tools for analysis and visualization
+import ase.geometry.rdf
+
+# tools to run the simulation
+import ase.io
+import ase.md
+import ase.visualize.plot
+import chemiscope
+import matplotlib.pyplot as plt
+
+# the usual suspects
+import numpy as np
+import torch
+
+from metatensor.torch import Labels, TensorBlock, TensorMap
+from metatensor.torch.atomistic import (
+    MetatensorAtomisticModel,
+    ModelCapabilities,
+    ModelMetadata,
+    ModelOutput,
+    NeighborListOptions,
+    System,
+)
+
+# Integration with ASE calculator for metatensor atomistic models
+from metatensor.torch.atomistic.ase_calculator import MetatensorCalculator
+
+
+# %%
+#
+# A Lennard-Jones Model
+# ---------------------
+#
+# The Lennard-Jones potential is a mathematical model to approximate the interaction
+# between a pair of neutral atoms or molecules. It is given by the equation:
+#
+# .. math:: V(r) = 4\epsilon \left[ \left( \frac{\sigma}{r} \right)^{12} - \left(
+#     \frac{\sigma}{r} \right)^{6} \right]
+#
+# where :math:`\epsilon` is the depth of the potential well, :math:`\sigma` is the
+# finite distance at which the inter-particle potential is zero, and :math:`r` is the
+# distance between the particles. The 12-6 form is chosen because the :math:`r^12` term
+# approximates the Pauli repulsion at short ranges, while the :math:`r^6` term
+# represents the attractive van der Waals forces. The :math:`r^{12}` is choosen becuase
+# it is just the square of the :math:`r^6` and therefore allows fast evaluation.
+#
+# A Lennard-Jones potential is well-suited for argon because it accurately represents
+# the van der Waals forces that dominate the interactions between argon atoms especially
+# as for all noble gases. This potential was used in one of the first MD simulations:
```

Comment:
```suggestion
# A Lennard-Jones potential is well-suited for argon because it accurately represents
# the van der Waals forces that dominate the interactions between argon atoms,
# as for all noble gases. This potential was used in one of the first MD simulations:
```